File: javascript/src/elements/missing/missing_attribute.ts
import { MissingValueElementBase } from './missing_value_element_base.js';

export class MissingAttribute extends MissingValueElementBase {
    get [Symbol.toStringTag]() {
        return 'MissingAttribute';
    }
    
    toString() {
        if (this._key === null)
            return `[object MissingAttribute]`;
        
        return `[object MissingAttribute key=${this._key}]`;
    }
}

File: ruby/lib/enolib/elements/missing/missing_attribute.rb
# frozen_string_literal: true

module Enolib
  class MissingAttribute < MissingValueElementBase
    def to_s
      if @key
        "#<Enolib::MissingAttribute key=#{@key}>"
      else
        '#<Enolib::MissingAttribute>'
      end
    end
  end
end
